File: Data Structures & Algorithms/search-insert-position/submission-1.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        int left = 0;
        int right = nums.length - 1;

        while(left <= right){
            int mid = left + (right - left) / 2;

            if(nums[mid] == target) return mid;
            else if (target < nums[mid]) right = mid - 1;
            else left = mid + 1;
        }

        // return (target > nums[mid]) ? left + 1 : right + 1;
        // System.out.println("Left Right: " + left + ", " + right);
        return left;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-1.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(heights) {
        let left = 0;
        let right = heights.length - 1;
        let max = 0;

        while(left < right){
            let height = Math.min(heights[left], heights[right]);
            let width = right - left;

            let area = height * width;

            max = Math.max(area, max);

            if(heights[left] < heights[right]){
                left++;
            }
            else{
                right--;
            }
        }

        return max;
    }
}
